### PR TITLE
[IOPAE-1407] Added PageHeader component to @io-ipatente/ui package

### PR DESCRIPTION
**Pull request by giuseppedipinto**
Added `PageHeader` ui component.

#### List of Changes
- added `PageHeader` ui component
- added `PageHeader` story
- added `PageHeader` unit tests

#### How Has This Been Tested?
Local `dev` mode.

#### Screenshots (if appropriate):
Screenshot 2024-10-03 alle 17 09 38

Screenshot 2024-10-03 alle 17 11 29
Screenshot 2024-10-03 alle 17 11 49

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: packages/ui/lib/components/page-header/Pageheader.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+  breadcrumbs?: UIPageHeaderBreadcrumb[];
+  /** Page Header description: if present, it is shown under the title */
+  description?: string;
+  /** Event triggered when user click on a breadcrumb with `routePath` defined */
+  onBreadcrumbClick?: (routePath: string) => void;
+  /** Page Header main title label */
+  title: string;
+}
+
+export interface UIPageHeaderBreadcrumbs {
```
Comment (gunzip) [inline packages/ui/lib/components/page-header/Pageheader.tsx:20]:
```suggestion
export interface UIPageHeaderBreadcrumbsProps {
```

File: packages/ui/lib/components/page-header/Pageheader.tsx
[the remark this answers is not in the document]
Comment (giuseppedipinto) [reply packages/ui/lib/components/page-header/Pageheader.tsx:1]: Sure it'a a typo, i 'll fix asap, thanks

File: packages/ui/lib/components/page-header/Pageheader.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+}: UIPageHeaderProps) => (
+  <Stack
+    bgcolor={PAGE_HEADER_BG_COLOR}
+    data-testid={DATA_TEST_ID_PREFIX}
+    paddingX={2}
+    paddingY={3}
+    spacing={1}
+  >
+    {breadcrumbs && (
+      <UIPageHeaderBreadcrumbs
```
[the remark this answers is not in the document]
Comment (giuseppedipinto) [reply packages/ui/lib/components/page-header/Pageheader.tsx:53]: It's a very interesting idea, thanks!

File: packages/ui/lib/components/page-header/__tests__/PageHeader.test.tsx [hunk 3]
```diff
@@ -0,0 +1,105 @@
+import "@testing-library/jest-dom";
+import { fireEvent, render, screen } from "@testing-library/react";
+import { expect, it, vi } from "vitest";
+
+import {
+  DATA_TEST_ID_PREFIX,
+  UIPageHeader,
+  UIPageHeaderBreadcrumb,
+} from "../Pageheader";
```
[the remark this answers is not in the document]
Comment (giuseppedipinto) [reply packages/ui/lib/components/page-header/__tests__/PageHeader.test.tsx]: done

File: packages/ui/lib/components/page-header/Pageheader.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (giuseppedipinto) [reply packages/ui/lib/components/page-header/Pageheader.tsx]: 9255b16

File: packages/ui/lib/components/page-header/Pageheader.tsx (hunk 1 above)
Comment (giuseppedipinto) [reply packages/ui/lib/components/page-header/Pageheader.tsx]: 9255b16

**Review by gunzip — APPROVED** [review]

**Review by adelloste — APPROVED** [review]

**Pull request merged**